### PR TITLE
APIM-2022-08-01-draft

### PR DESCRIPTION
**Pull request by hoonality**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [x] Other, please clarify - adding definitions for a new entity
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month. **Targeting mid to late October.**
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month. **Targeting mid to late October.**
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

**Added notes**
Notes for context if helpful:
* There is a plan to have a 2022-08-01 stable version to be added soon by our team to the main repo (likely will be called something like apim-2022-08-01).
* For this new feature, I created an Azure/apim-2022-08-01-draft branch as well as a fork and branch that is intended to merge in with the apim-2022-08-01 stable version once it is available in the main repo.
* Because of our timeline, I am hoping that by initiating a review here will help with the turnaround time for ARM review.
  * The bits that I changed for the new feature are in these commits only: 
    * https://github.com/Azure/azure-rest-api-specs/pull/20728/commits/79fb17d4c69786feabde8a7d83b0f2d32b5075f8
    * https://github.com/Azure/azure-rest-api-specs/pull/20728/commits/8776156befc0b6948b52793a25b5f81bd2256bf5 
### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2022-08-01/apimanagement.json [hunk 1]
```diff
@@ -0,0 +1,867 @@
...
+      "name": "operationId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "Operation identifier within an API. Must be unique in the current API Management service instance.",
+      "minLength": 1,
+      "maxLength": 80,
+      "x-ms-parameter-location": "method"
+    },
+    "ResolverIdParameter": {
```
Comment (raosuhas) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2022-08-01/apimanagement.json:543]: >ResolverIdParameter [](http://example.com/codeflow?start=5&length=19)

Why is this added here ? Does not look like this is used anywhere in this file?

**Comment by raosuhas** [comment]
>       "head": {

have you tested this at all ? AFAIK ARM does not support the HEAD function in REST

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2022-08-01/apimapis.json:1864 in 7b927b5. [](commit_id = 7b927b5f2c7725840b445ba2a51a0b72f9c46f0d, deletion_comment = False)

**Comment by raosuhas** [comment]
>         },

avoid returning count as explained here : 

[https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#collections](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#collections)

<a name="collections-avoid-count-property> ⚠️ YOU SHOULD NOT return a count of all objects in the collection as this may be expensive to compute.

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2022-08-01/definitions.json:5575 in 7b927b5. [](commit_id = 7b927b5f2c7725840b445ba2a51a0b72f9c46f0d, deletion_comment = False)

**Comment by raosuhas** [comment]
>         "value": "<http-data-source><http-request><set-method>GET</set-method><set-backend-service base-url=\"https://some.service.com\" /><set-url>/api/users</set-url></http-request></http-data-source>"

how large can this value get ?

please note the size limitations for ARM payload here : 

[https://github.com/Azure/azure-resource-manager-rpc/blob/022630d456f8a92999040b0193929807815e7246/v1.0/common-api-details.md#max-request-body-size](https://github.com/Azure/azure-resource-manager-rpc/blob/022630d456f8a92999040b0193929807815e7246/v1.0/common-api-details.md#max-request-body-size)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2022-08-01/examples/ApiManagementCreateGraphQLApiResolverPolicy.json:14 in 7b927b5. [](commit_id = 7b927b5f2c7725840b445ba2a51a0b72f9c46f0d, deletion_comment = False)

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2022-08-01/apimanagement.json (hunk 1 above)
Comment (raosuhas) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2022-08-01/apimanagement.json:543]: should this be in the definitions file instead ?

Comment (hoonality) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2022-08-01/apimanagement.json:543]: I am not sure about the origins of the pattern but our team seems to add parameters to the apimanagement.json file and refer to them in the apimapis.json. I can try to dig and try to find out why if moving these to the definitions files is preferred.

**Comment by hoonality** [comment]
will close and reopen PR as there seems to be a large difference between the new release branch for 2022-08-01 and the draft originally setup
